### PR TITLE
Fix incorrect 'drag_from' assignment of GraphNodes

### LINKED ISSUE #59843: GraphNode offset incorrectly changes if selected from code while dragging another node
### Godot version

v3.4.4.stable.official [419e713a2]

### System information

Windows 10

### Issue description

```GraphNode.offset``` is set incorrectly if you select it from code while already dragging another node:
example_gif

### Steps to reproduce

1. Start dragging ```GraphNode1```.
2. Set ```GraphNode2.selected = true```, for example from ```GraphNode1.offset_changed()``` signal callback.
3. ```GraphNode2.offset``` is set incorrectly.

### Minimal reproduction project

[GraphEditTest.zip](https://github.com/godotengine/godot/files/8405225/GraphEditTest.zip)

### PR DESCRIPTION
**Pull request by sriramun**
Fixes #59843

### CODE REVIEW

File: scene/gui/graph_node.cpp [hunk 1]
```diff
@@ -596,14 +596,18 @@ bool GraphNode::is_selected() {
...
-		drag_from = get_offset();
-	} else {
+	if (!p_drag) {
 		emit_signal("dragged", drag_from, get_offset()); //useful for undo/redo
 	}
+	drag_from = get_offset();
 }
 
 Vector2 GraphNode::get_drag_from() {
+
```
Comment (YeldhamDev) [inline scene/gui/graph_node.cpp:606]:
```suggestion
```

This empty line is unnecessary.

**Review by YeldhamDev — CHANGES_REQUESTED** [review]

File: scene/gui/graph_node.cpp [hunk 2]
```diff
@@ -596,14 +596,17 @@ bool GraphNode::is_selected() {
...
-		drag_from = get_offset();
-	} else {
+	if (!p_drag) {
 		emit_signal("dragged", drag_from, get_offset()); //useful for undo/redo
 	}
+	drag_from = get_offset();
 }
 
 Vector2 GraphNode::get_drag_from() {
+	if (drag_from == Vector2(0,0)) {
```
Comment (YeldhamDev) [inline scene/gui/graph_node.cpp:606]:
```suggestion
	if (drag_from == Vector2(0, 0)) {
```

**Review by YeldhamDev — CHANGES_REQUESTED** [review]

**Review by YeldhamDev — APPROVED** [review]
It fixes the bug, but I don't have much experience with `GraphNode`, so I want the opinion of another maintainer as well.
And if everything is fine, it should  also be ported to `master`, as the bug happens there.

**Review by Geometror — COMMENTED** [review]
I can confirm that this fixes the issue. However, there are a couple of ways to fix this, I'm not sure if this is the best one (although I'm not completely opposed to merging). It really depends on how we want the API to work. For now, I think it should probably be done in GraphEdit since that class is actually responsible for the whole dragging logic. (GraphNode's set_drag and get_drag_from are not exposed and only used by GraphEdit). So here is what we could do: (No guarantees that the following will work) GraphEdit can update `drag_from` via `set_drag` when several nodes are selected and the user begins the drag.
As an alternative, just inserting
```cpp
drag_from = get_position_offset();
```
in `set_selected` also fixes the issue.

**Comment by YuriSizov** [comment]
> although I'm not completely opposed to merging

For the record, I am opposed. I haven't looked deep into it, but I don't like that this PR compares things to `Vector2(0, 0)` and that the getter is changing the value internally. Getters should be `const` or at least avoid modifying the internal field. So this doesn't feel like a good solution.

**Comment by YuriSizov** [comment]
I'll close this based on my comment above. A different solution is needed. And we should fix it in `master` too.

**Pull request closed without merging**